File: heap_dynamic.java
import java.util.ArrayList;

public class HeapDynamicExample {

    public static void main(String[] args) {
        // Heap-Dynamic: uses ArrayList, size can grow or shrink
        ArrayList<Integer> dynamicList = new ArrayList<>();

        // Add elements at runtime
        dynamicList.add(10);
        dynamicList.add(20);
        dynamicList.add(30);
        dynamicList.add(40);

        // Modify contents
        dynamicList.remove(1);  // removes value at index 1 (20)
        dynamicList.add(50);

        // Display the dynamic list
        System.out.println("Heap-Dynamic Array:");
        for (Integer value : dynamicList) {
            System.out.print(value + " ");
        }
    }
}

File: heap_dynamic.py
def heap_dynamic():
    arr = []  # empty list, heap-allocated

    # Add elements dynamically
    arr.append(10)
    arr.append(20)
    arr.append(30)
    arr.append(40)

    # Modify content
    arr[1] = 25  # change 20 to 25
    arr.pop()  # remove last element (40)

    # Add more elements
    arr.extend([50, 60])

    # Display the list
    print("Heap-Dynamic Array:")
    print(arr)


heap_dynamic()
